lib/archive/archiver.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"io"
+	"net/url"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	// ExcludeString defines the string that specifies a path exclusion.
+	ExcludeString = "exclude"
+	// IncludeString defines the string that specifies a path inclusion.
+	// Note: An included path that sits under an excluded path should be included.
+	// For example, if / is excluded, and /files is included, /files should  be added to the archive.
+	IncludeString = "include"
+	// RebaseString defines the path that was stripped when copying from a volume vmdk.
+	// This is necessary as the volume may be mounted deep in the container filesystem,
+	// but files on the volume are seen as living in the filesystem root when being diffed.
+	RebaseString = "rebase"
+)
+
+// PathSpec describes rules for handling specified paths during archival
+type PathSpec struct {
+	Inclusions map[string]struct{}
+	Exclusions map[string]struct{}
+	RebasePath string
+}
+
+// Archiver defines an API for creating archives consisting of data that exists between a child and parent
+// layer, as well as unpacking archives to container filesystems
+type Archiver interface {
+
+	// ReadArchive reads the delta between child and parent image layers, returning
+	// the difference as a tar archive.
+	//
+	// store - the image store containing the two layers
```

Comment:
s/image/device/ since we will be using it to read more than just the image store.